### PR TITLE
Add a diff method for coordinates

### PR DESCRIPTION
**Pull request by cquiroz**
The current offset functions on `Coordinates` are for a planar geometry but we often need them on a spherical one.
This PR removes the old ones and adds one taken from the OT

### CODE REVIEW

File: modules/core/shared/src/main/scala/lucuma/core/math/Coordinates.scala [hunk 1]
```diff
@@ -11,39 +11,50 @@ import lucuma.core.syntax.all._
...
+      val acosValue = acos(cosd)
+      if (acosValue.isNaN) 0.0 else acosValue
     }
 
-  /**
-   * Offset these `Coordinates` by the given deltas. If the declination crossed a pole the right
-   * ascension will have been flipped 180°.
-   * @group Operations
-   */
-  def offset(dRA: HourAngle, dDec: Angle): Coordinates =
```
Comment (swalker2m) [inline modules/core/shared/src/main/scala/lucuma/core/math/Coordinates.scala:38]: I think `offset` and `offsetWithCarry` are correct.  It adds an offset in `HourAngle` so it's correct at any declination, though the actual angular separation will differ according to declination.  At any rate, maybe we don't need this method at the moment but I would keep it and `offsetWithCarry`.

**Review by swalker2m — COMMENTED** [review]
Looks great, thanks for fixing this!  I would just vote to keep the `offset` methods as well.

File: modules/core/shared/src/main/scala/lucuma/core/math/Coordinates.scala (hunk 1 above)
Comment (swalker2m) [reply modules/core/shared/src/main/scala/lucuma/core/math/Coordinates.scala:38]: I guess I could be swayed that this calculation, though correct, is never going to be useful, if anyone disagrees.

Comment (cquiroz) [reply modules/core/shared/src/main/scala/lucuma/core/math/Coordinates.scala:38]: I thought about keeping but renaming it as `planarOffset` and add another `offsetBy(posAngle, offset): Coordinates` method

**Review by swalker2m — APPROVED** [review]
Looks good. :+1:

**Pull request merged**